MatrixSDK/Data/RoomList/CoreData/MXCoreDataRoomListDataFetcher.swift

```diff
@@ -380,3 +389,70 @@ extension MXCoreDataRoomListDataFetcher: NSFetchedResultsControllerDelegate {
     }
     
 }
+
+//  MARK: - RoomSummaryForTotalCounts
+
+/// Room summary implementation specific to total counts calculation.
+private class RoomSummaryForTotalCounts: NSObject, MXRoomSummaryProtocol {
+
+    var roomId: String = ""
+    var roomTypeString: String?
+    var roomType: MXRoomType = .room
+    var avatar: String?
+    var displayname: String?
+    var topic: String?
+    var creatorUserId: String = ""
+    var aliases: [String] = []
+    var joinRule: String? = kMXRoomJoinRuleInvite
+    var membership: MXMembership = .unknown
+    var membershipTransitionState: MXMembershipTransitionState = .unknown
+    var membersCount: MXRoomMembersCount = MXRoomMembersCount(members: 0, joined: 0, invited: 0)
+    var isConferenceUserRoom: Bool = false
+    var hiddenFromUser: Bool = false
+    var storedHash: UInt = 0
+    var lastMessage: MXRoomLastMessage?
+    var isEncrypted: Bool = false
+    var trust: MXUsersTrustLevelSummary?
+    var localUnreadEventCount: UInt = 0
+    var notificationCount: UInt
+    var highlightCount: UInt
+    var hasAnyUnread: Bool {
+        return localUnreadEventCount > 0
+    }
+    var hasAnyNotification: Bool {
+        return notificationCount > 0
+    }
+    var hasAnyHighlight: Bool {
+        return highlightCount > 0
+    }
+    var isDirect: Bool {
+        return isTyped(.direct)
+    }
+    var directUserId: String?
+    var others: [String: NSCoding]?
+    var favoriteTagOrder: String?
+    var dataTypes: MXRoomSummaryDataTypes
+
+    func isTyped(_ types: MXRoomSummaryDataTypes) -> Bool {
+        return (dataTypes.rawValue & types.rawValue) != 0
+    }
+
+    var sentStatus: MXRoomSummarySentStatus
+    var spaceChildInfo: MXSpaceChildInfo?
+    var parentSpaceIds: Set<String> = []
+
+    /// Initializer with a dictionary obtained by a fetch request, with result type `.dictionaryResultType`. Only parses some properties.
+    /// - Parameter dictionary: Dictionary object representing an `MXRoomSummaryMO` instance
+    init(withDictionary dictionary: [String: Any]) {
+        dataTypes = MXRoomSummaryDataTypes(rawValue: dictionary[kPropertyNameDataTypesInt] as? Int ?? 0)
+        sentStatus = MXRoomSummarySentStatus(rawValue: dictionary[kPropertyNameSentStatusInt] as? UInt ?? 0) ?? .ok
+        notificationCount = dictionary[kPropertyNameNotificationCount] as? UInt ?? 0
+        highlightCount = dictionary[kPropertyNameHighlightCount] as? UInt ?? 0
```

Comment:
All looks good to me. My only thought is whether these dictionary values should be cast to the same Int types that is used in the model definition?
e.g. `Int64` for `s_dataTypesInt` and `Int16` for `sentStatusInt`, `notificationCount` and `highlightCount`?

Reply:
Does not matter much in this case, updated anyway.